superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -503,7 +503,6 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
     setImportingModal(false);
     setPasswords({});
     setConfirmedOverwrite(false);
-    if (onDatabaseAdd) onDatabaseAdd();
```

Comment:
I'm curious about the reason for the deletion here, it looks like the `close` method will be called in multiple places, not just the model closing.

Reply:
Yeah, sure.
Calling `onDatabaseAdd` in the `onClose` causes the callback to be fired when it shouldn't.
The `onClose` is called, for instance, if you hit in the overlay.
That causes (in master) the following:

https://user-images.githubusercontent.com/17252075/171906617-319a39e7-647c-418d-bae6-8027a6e805fe.mov

The only places that should be called is when a change was actually made (which was done already in almost all places, causing a double fetch/rerender).
The only place that uses this function is `DatabaseList`, which only does a refresh.